src/generateDescriptions.ts

```diff
@@ -0,0 +1,91 @@
+import { promises as fs } from "fs";
+import * as path from "path";
+import { fileURLToPath } from "url";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+function extractSummary(markdown: string): string {
+  // Remove frontmatter (--- at the beginning)
+  markdown = markdown.replace(/^---[\s\S]+?---\n/, "");
+
+  // Normalize line breaks by collapsing consecutive newlines into a single space
+  const normalizedText = markdown
+    .split("\n")
+    .map((line) => line.trim())
+    .filter(
+      (line) =>
+        line &&
+        !line.startsWith("#") &&
+        !line.startsWith(">") &&
+        !line.startsWith("{{"),
+    )
+    .join(" ");
+
+  // Extract the first sentence (ending in . ! or ?)
+  const sentenceMatch = normalizedText.match(
+    /(.*?(?<!\.\s(?:i\.e|e\.g))([.!?]))(?:\s|$)/i,
```

Comment:
This doesn't really solve the "e.g." case, as replacing this with `/(.*?[.!?])(?=\s|$)/` doesn't change anything. I'm okay with deferring that issue for now, but in that case we should revert this regex, to prevent giving false sense that this solves the case when it does not.